test/hotspot/jtreg/runtime/MemProfiler/MemProfilingWithGC.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2021, Alibaba Group Holding Limited. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+/*
+ * @test MemProfilingWithGC
+ * @bug 8264682
+ * @summary test combinations of various GC and -XX:+MemProfiling
+ * @library /test/lib
+ * @run driver MemProfilingWithGC
+ */
+
+import jdk.test.lib.process.ProcessTools;
+import jdk.test.lib.process.OutputAnalyzer;
+
+public class MemProfilingWithGC {
+    public static void main(String... args) throws Exception {
+        final String GCs[] = new String[] {
+            "UseG1GC",
+            "UseSerialGC",
+            "UseParallelGC",
+            "UseShenandoahGC",
+            "UseZGC",
+            "UseEpsilonGC",
```

Comment:
You have to allow for the fact that these GC's may not have all been built into the JVM under test.